website/content/platform/development/contributing/how-to/add_endpoint_to_existing_provider.md

```diff
@@ -0,0 +1,802 @@
+| XAU             | HKD                |   16341.8   | 16310      | 16341.9    | 16276.4    |     1665 |        16307 |   34.75  |      0.002131    | 15891.1    | 15452.8    |   16306.3   |  14238     | 2024-03-04 05:57:11   |
+| XAG             | USD                |      23.299 |    23.1091 |    23.3062 |    23.0172 |     2074 |           23 |    0.299 |      0.013       |    22.7862 |    23.4349 |      26.035 |     20.005 | 2024-03-04 05:56:41   |
+| XAG             | GBP                |      18.26  |    18.21   |    18.26   |    18.14   |      413 |           18 |    0.26  |      0.0144444   |    17.9988 |    18.5021 |      20.67  |     16.81  | 2024-03-04 05:24:10   |
+| XAG             | EUR                |      21.36  |    21.32   |    21.37   |    21.2087 |     1079 |           21 |    0.36  |      0.0171429   |    21.0393 |    21.4906 |      23.64  |     18.97  | 2024-03-04 05:30:10   |
+| XAG             | HKD                |     181.237 |   180.881  |   181.399  |   180.124  |     1596 |          180 |    1.237 |      0.0068722   |   178.342  |   181.815  |     204.411 |    157.209 | 2024-03-04 05:30:10   |
+
+## Write Tests
+
+We'll need to create a unit test for the FMP provider, and then integration tests for the Python interface and Fast API. It's as simple as creating a new router function was, copying and pasting.
+
+### Unit Test
+
+This is located in the `openbb-fmp` extension:
+
+```console
+~/OpenBBTerminal/openbb_platform/providers/fmp/tests/test_fmp_fetchers.py
+```
+
+- Import the new fetcher with the rest of the imports (keep them alphabetically sorted).
+- Copy and paste the last test function in the file.
+
+```python
+@pytest.mark.record_http
+def test_fmp_currency_snapshots_fetcher(credentials=test_credentials):
+    params = {
+        "base": "XAU",
+        "quote_type": "indirect",
+        "counter_currencies": "USD,EUR,GBP,JPY,HKD,AUD,CAD,CHF,SEK,NZD,SGD",
+    }
+
+    fetcher = FMPCurrencySnapshotsFetcher()
+    result = fetcher.test(params, credentials)
+    assert result is None
+```
+
+- Navigate to the path above and enter: `pytest test_fmp_fetchers.py --record http --record-no-overwrite`
+
+This will generate a new file:
+
+```console
+~/OpenBBTerminal/openbb_platform/providers/fmp/tests/record/test_fmp_currency_snapshots_fetcher.yaml
+```
+
+Check the file for any obvious errors, like a bad HTTP request status code.
+
+### Integration Tests
+
+The Python interface and Fast API each require a new integration test. Again, emulate an existing test and make sure to declare all parameters available to each provider.
+
+#### API
+
+Open the file below, and go to the last test in the file.
+
+```console
+~/OpenBBTerminal/openbb_platform/extensions/currency/integration/test_currency_api.py
+```
+
+We can copy this one:
```

Comment:
They can leverage our QA framework command to create a test instead of copy/pasting.